lib/common/acl.c

```diff
@@ -434,22 +474,23 @@ xml_acl_filtered_copy(const char *user, xmlNode *acl_source, xmlNode *xml,
             for(lpc = 0; lpc < max; lpc++) {
                 xmlNode *match = getXpathResult(xpathObj, lpc);
 
-                crm_trace("Purging attributes from %s", acl->xpath);
                 if (__xml_purge_attributes(match) == FALSE && match == target) {
-                    crm_trace("No access to the entire document for %s", user);
+                    crm_trace("ACLs deny user '%s' access to entire XML document",
+                              user);
                     freeXpathObject(xpathObj);
                     return TRUE;
                 }
             }
-            crm_trace("Enforced ACL %s (%d matches)", acl->xpath, max);
+            crm_trace("Enforced deny ACL %s (%d match%s)",
+                      acl->xpath, max, ((max == 1)? "" : "es"));
```

Comment:
3. is it time for `es_if_plural` macro addition?

- not that it'd be important, just recalled from the past

Reply:
6. minor, perhaps "Enforced deny '%s' user ACL rule for %s ..."

Also, there's some purely linguistic reasoning around "ACL" and
"ACLs".  Traditional view seems to be that ACL, access control list
(singular), is attached to the object, meaning the list itself contains
just pairs "subject, perms".  Set of these lists for multiple/all
objects is then naturally "ACLs" (plural).

All other usages of ACL/ACLs seemed fine, but "deny ACL" seems strange,
since we want to talk about a single rule from ACL for particular
"object" (defined in a query language) that generally contains various
perms for various users (even if encoded in a scattered way), hence
not just "deny" ones.